src/app/context/AccountContext.tsx

```diff
@@ -77,11 +77,9 @@ export function AccountProvider({ children }: { children: React.ReactNode }) {
     });
   };
 
-  const setAccountId = (id: string, willFetch: boolean) => {
+  const setAccountId = (id: string) => {
     setAccount({ id });
-    if (willFetch) {
-      setBalanceLoading(true);
-    }
+    setBalanceLoading(true);
```

Comment:
TBH I don't get this line of code: the balanceLoading state is set to `true` in the first line of `fetchAccountInfo` and set to `false` before the return statement. Why do we need to set it here? And if so, the only places where it is set to false is within the `fetchAccountInfo` function as well. So if we set it to `true` outside, but never call it, it will never be set to `false`.

Reply:
Also why is it set to false in the callback of `fetchAccountInfo` if it is being set after the await function anyways.

Reply:
@reneaaron I don't understand this callback function. The only place in the whole repo where `await api.swr.getAccountInfo` is called is inside `fetchAccountInfo` of AccountContext. Inside of `api.swr.getAccountInfo` the callback is being called just before the `return resolve(account)`. To me it seems it would make no difference if this callback is removed entirely and the functions of this callback

``` 
    const callback = (accountRes: AccountInfo) => {
      setAccount(accountRes);
      updateAccountBalance(accountRes.balance, accountRes.currency);
      setBalanceLoading(false);
    };
```

would just be called after the `await`.

Reply:
@lujakob Thanks for your review! 

> Edit: I think I found it. Probably you need it because if this delay of calling setAccountId and fetchAccountInfo

Exactly. I'll add a comment. 

The callback is also called here. I think it's purpose is to update the UI from the cache (while still revalidating in the background and then updating again as the new data is loaded. 
https://github.com/11WALTER11/lightning-browser-extension/blob/update/src/common/lib/api.ts#L69